### PR TITLE
Update progress indicators

### PR DESCRIPTION
**Pull request by microbit-grace**
- Make loading spinners/bars green
- Update training in progress spinner to loading bar
- Add isLoading state for StandardButton that uses same spinner component

https://microbit-global.monday.com/boards/1356069004/pulses/1356427969

### CODE REVIEW

**Comment by microbit-grace** [comment]
Preview: https://progress-indicators.ml-trainer.pages.dev/

**Comment by microbit-grace** [comment]
Things to fix: 
- [ ] Reset training loading bar after training is complete
- [ ] Update credit link to Github link instead of link to examples
- [ ] Remove use of wait (if the reason for it is not obvious)

File: src/script/ml.ts [hunk 1]
```diff
@@ -64,7 +61,7 @@ function createModel(): LayersModel {
   return model;
 }
 
-export async function trainModel(): Promise<boolean> {
+export async function trainModel() {
```
Comment (microbit-robert) [inline src/script/ml.ts:64]: We may as well keep the type and change it to how it use to be.
```suggestion
export async function trainModel(): Promise<void> {
```

Also need to change line 74 from `return false;` to `return;`

File: src/script/ml.ts [hunk 2]
```diff
@@ -105,32 +102,27 @@ export async function trainModel(): Promise<boolean> {
...
   nn.fit(tensorFeatures, tensorLabels, {
-    epochs: get(settings).numEpochs,
+    epochs: totalNumEpochs,
     batchSize: 16,
     validationSplit: 0.1,
-    callbacks: { onTrainEnd }, // onEpochEnd <-- use this to make loading animation
+    callbacks: {
+      onTrainEnd,
+      onEpochEnd: (epoch: number) => {
+        // epoch only increases up to totalNumEpochs - 1
```
Comment (microbit-robert) [inline src/script/ml.ts:115]: Can we change this comment to something like 'epochs indexed at 0', otherwise it looks like a bug.

**Review by microbit-robert — CHANGES_REQUESTED** [review]
Mostly looks good. I've added a couple of minor comments.

I don't think we should change StandardButton to take an isLoading prop. I think it's weird for a button to have a loading animation in it. We'll drop this shortly because these buttons will change to "connect" instead of loading.

**Review by microbit-robert — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**